cg_lims/EPPs/files/sample_sheet/models.py

```diff
@@ -0,0 +1,158 @@
+from genologics.lims import Process
+from typing import Optional
+import logging
+from enum import Enum
+
+LOG = logging.getLogger(__name__)
+
+
+class StrEnum(str, Enum):
+    def __str__(self) -> str:
+        return str.__str__(self)
+
+
+class IntEnum(int, Enum):
+    def __int__(self) -> int:
+        return int.__int__(self)
+
+
+class IndexSetup(IntEnum):
+    DUAL_INDEX: int = 2
+    SINGLE_INDEX: int = 1
+
+
+class SampleSheetHeader(StrEnum):
+    FILE_SECTION: str = "[Header]"
+    READS_SECTION: str = "[Reads]"
+    SETTINGS_SECTION: str = "[Sequencing_Settings]"
+    BCL_SETTINGS_SECTION: str = "[BCLConvert_Settings]"
+    BCL_DATA_SECTION: str = "[BCLConvert_Data]"
+
+
+class NovaSeqXRun:
+    process_id: Process
+    run_name: str
+    file_format: int = 2
+    instrument_type: str
+    instrument_platform: str
+    index_orientation: str
+    read_1_cycles: int
+    read_2_cycles: Optional[int]
+    index_1_cycles: Optional[int]
+    index_2_cycles: Optional[int]
+    override_cycles: Optional[bool] = True
+    bclconvert_software_version: str
+    bclconvert_app_version: Optional[str]
+    fastq_compression_format: str
+    trim_adapters: Optional[bool] = False
+
+    def __init__(self, process: Process):
+        self.process: Process = process
+        self.run_name: str = process.udf.get("Experiment Name")
+        self.instrument_type: str = "NovaSeqxPlus"
+        self.instrument_platform: str = "NovaSeqXSeries"
+        self.index_orientation: str = "Forward"
+        self.read_1_cycles: int = process.udf.get("Read 1 Cycles")
+        self.read_2_cycles: int = process.udf.get("Read 2 Cycles")
+        self.index_1_cycles: int = process.udf.get("Index Read 1")
+        self.index_2_cycles: int = process.udf.get("Index Read 2")
+        self.bclconvert_software_version: str = process.udf.get("BCLConvert Software Version")
+        self.fastq_compression_format: str = process.udf.get("Compression Format")
+
+    def create_head_section(self) -> str:
+        """Return the [Head] section of the sample sheet."""
```

Comment:
```suggestion
        """Return the header section of the sample sheet."""
```
Or is it the file section?